### PR TITLE
approver-policy docs

### PR DESCRIPTION
**Pull request by JoshVanL**
Adds docs for approver-policy.

Adds example policy resources.

Adds generated API docs.

/assign @irbekrm 

branched from #26 which should be merged first.

```release-notes
NONE
```

### CODE REVIEW

**Comment by JoshVanL** [comment]
`Error: looks like "https://charts.jetstack.io" is not a valid chart repository or cannot be reached: Get "https://charts.jetstack.io/index.yaml": dial tcp: lookup charts.jetstack.io on 10.8.240.10:53: read udp 10.8.4.18:45497->10.8.240.10:53: i/o timeout`

:cry:

**Review by irbekrm — CHANGES_REQUESTED** [review]
This makes sense to me in general, added a couple suggestions.

I see that the module still needs to be renamed `policy-approver` -> `approver-policy`

File: Makefile [hunk 1]
```diff
@@ -21,6 +21,8 @@ HELM_VERSION ?= 3.6.3
 KUBEBUILDER_TOOLS_VERISON ?= 1.22.0
 K8S_CLUSTER_NAME ?= policy-approver
 
+GOMARKDOC_FLAGS=--include-unexported --format github --repository.url "https://github.com/cert-manager/policy-approver" --repository.default-branch master --repository.path /
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply Makefile:24]: Done in #33

File: Makefile [hunk 2]
```diff
@@ -47,17 +45,14 @@ vet: ## Run go vet against code.
 	go vet ./...
 
 .PHONY: lint
-lint: ## Run linters against code.
+lint: $(BINDIR)/gomarkdoc ## Run linters against code.
 	./hack/verify-boilerplate.sh
+	@$(BINDIR)/gomarkdoc --check $(GOMARKDOC_FLAGS) --output docs/api/api.md github.com/cert-manager/policy-approver/pkg/apis/policy/v1alpha1 || (echo "docs are not up to date; run 'make generate' and commit the result" && exit 1)
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply Makefile:50]: Done in #33

File: Makefile [hunk 3]
```diff
@@ -79,8 +74,13 @@ smoke: demo ## create cluster, deploy policy-approver, run smoke tests
 	REPO_ROOT=$(shell pwd) ./hack/ci/run-smoke-test.sh
 	REPO_ROOT=$(shell pwd) ./hack/ci/delete-cluster.sh
 
+.PHONY: docs/api/api.md
+docs/api/api.md: $(BINDIR)/gomarkdoc
+	mkdir -p docs/api
+	$(BINDIR)/gomarkdoc $(GOMARKDOC_FLAGS) --output $@ github.com/cert-manager/policy-approver/pkg/apis/policy/v1alpha1
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply Makefile:80]: Done in #33 
/close

File: docs/api/api.md [hunk 4]
```diff
@@ -0,0 +1,714 @@
...
+import "github.com/cert-manager/policy-approver/pkg/apis/policy/v1alpha1"
+```
+
+\+k8s:openapi\-gen=true \+k8s:deepcopy\-gen=package\,register \+groupName=policy\.cert\-manager\.io
+
+## Index
+
+- [Variables](<#variables>)
+- [func Resource(resource string) schema.GroupResource](<#func-resource>)
+- [func addKnownTypes(scheme *runtime.Scheme) error](<#func-addknowntypes>)
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply docs/api/api.md:15]: Thanks, done :+1:

**Comment by irbekrm** [comment]
Thanks Josh, this looks good

/lgtm
/approve

**Pull request merged**